File: app/src/main/java/br/com/gsas/app/picpay/Database/DatabaseManager.java
package br.com.gsas.app.picpay.Database;

import android.arch.persistence.room.Room;
import android.content.Context;

import br.com.gsas.app.picpay.DAO.CartaoDAO;
import br.com.gsas.app.picpay.DAO.FeedDAO;

public class DatabaseManager {

    private AppDatabase dbInstance;

    public static DatabaseManager getInstance(Context context) {

        return new DatabaseManager(context);
    }

    private DatabaseManager(Context context) {

        dbInstance = Room.databaseBuilder(context, AppDatabase.class, "picpay.sqlite").build();
    }

    public CartaoDAO getCartaoDAO(){

        return dbInstance.cartaoDAO();
    }

    public FeedDAO getFeedDAO(){

        return dbInstance.feedDAO();
    }

}

File: app/src/main/java/com/v1pi/picpay_teste/Database/DatabaseManager.kt
package com.v1pi.picpay_teste.Database

import android.arch.persistence.room.Database
import android.arch.persistence.room.Room
import android.arch.persistence.room.RoomDatabase
import android.content.Context
import com.v1pi.picpay_teste.Dao.CreditCardDao
import com.v1pi.picpay_teste.Domains.CreditCard

@Database(entities = [CreditCard::class], version = 2)
abstract class DatabaseManager : RoomDatabase() {

    abstract fun creditCardDao() : CreditCardDao

    companion object {
        private var INSTANCE : DatabaseManager? = null

        fun getInstance(context: Context) : DatabaseManager? {
            if(INSTANCE == null) {
                synchronized(DatabaseManager::class) {
                    INSTANCE = Room.databaseBuilder(context.applicationContext, DatabaseManager::class.java, "app_database").build()
                }
            }

            return INSTANCE
        }

        fun destroyInstance() {
            INSTANCE?.close()
            INSTANCE = null
        }
    }
}